src/js/components/header/search.js

```diff
@@ -1,38 +1,52 @@
-var $ = require('jquery');
+var select = require('../../utils/select');
+var addClass = require('../../utils/add-class');
+var removeClass = require('../../utils/remove-class');
+var dispatch = require('../../utils/dispatch');
 
-var $searchForm = $('.js-search-form');
-var $searchButton = $('.js-search-button');
+var searchForm = select('.js-search-form')[ 0 ];
+var searchButton = select('.js-search-button')[ 0 ];
+var searchButtonContainer = select('.js-search-button-container')[ 0 ];
+var searchDispatcher;
 
-$searchButton.on('click touchstart', function (e) {
-  e.preventDefault();
+if (searchButton && searchForm) {
+  dispatch(searchButton, 'click touchstart', searchButtonClickHandler);
+}
 
-  if (isOpen($searchForm)) {
+function searchButtonClickHandler (event) {
+  if (isOpen(searchForm)) {
     closeSearch();
   } else {
     openSearch();
-    $('html').on('click touchstart', function (e) {
-      var $target = $(e.target);
-      if ($target.parents('.js-search-form').length ||
-          $target.parents('.js-search-button-container').length) {
-        console.log('inside the item');
-      } else {
-        closeSearch();
-        $('html').off('click touchstart');
-      }
-    });
+    searchDispatcher = dispatch(document.body, 'click touchstart', searchOpenClickHandler);
+  }
+
+  return false;
+}
+
+function searchOpenClickHandler (event) {
+  var target = event.target;
+  if (! searchFormContains(target)) {
```

Comment:
There for readability & obviousness, because (in my experience) people often miss the exclamation mark in `(!something)`.

That said, this is the only usage of that method. I was considering naming it `searchFormDoesNotContain()` but that seemed rather unwieldy. What do you think?